client/reader/full-post/index.jsx

```diff
@@ -233,6 +232,28 @@ FullPostView = React.createClass( {
 		/*eslint-enable react/no-danger*/
 	},
 
+	renderGalleryScripts: function() {
+		/**
+		 * NOTE: Reuse methods from components/tinymce/plugins/wpcom-view/plugin.js
+		 * to parse content and apply divs/scripts, below is a placeholder for now
+		 */
+		return React.createElement( GalleryView, {
+			siteId: 18561139,
+			content: '[gallery ids=\"2247,2246,2245,2250,2251,2255,2252\" type=\"slideshow\"]'
+		} );
+	},
+
+	renderFullPostContent: function( content ) {
+		return (
+			<div>
+				<EmbedContainer>
+					<div className="reader__full-post-content" _dangerouslySetInnerHTML={ { __html: content } } />
+				</EmbedContainer>
+				{ /data-gallery/.test( content ) ? this.renderGalleryScripts() : null }
```

Comment:
i don't think we want to do this on every render call. probably either do it after mount / update, or up in post-normalizer.